core-primitives/enclave-api/src/remote_attestation.rs

```diff
@@ -157,14 +158,15 @@ mod impl_ffi {
 					url.len() as u32,
 					unchecked_extrinsic.as_mut_ptr(),
 					unchecked_extrinsic.len() as u32,
+					&mut unchecked_extrinsic_size as *mut u32,
 					skip_ra.into(),
 				)
 			};
 
 			ensure!(result == sgx_status_t::SGX_SUCCESS, Error::Sgx(result));
 			ensure!(retval == sgx_status_t::SGX_SUCCESS, Error::Sgx(retval));
 
-			Ok(unchecked_extrinsic)
+			Ok(Vec::from(&unchecked_extrinsic[..unchecked_extrinsic_size as usize]))
```

Comment:
I think this needs a check that `unchecked_extrinsic_size < unchecked_extrinsic.len()` to prevent panics. This true for all the following usages too.

Reply:
I think this is implicitly guaranteed by the ffi impl, isn't it

If it's out of index, it should have error'ed out within `write_slice_and_whitespace_pad`

Reply:
Hmm, yes and no. If we have implemented everything correctly, yes, but the compiler can't make any checks across the ffi-boundaries. So I prefer being defensive here, as it just needs an error of some programmer in the future to introduce potential panics here.

Sorry, for being nitpicky here. 🙏

Reply:
Hmm, on the other hand the panic is on the untrusted side, so returning an error or panicking here, is essentially the same from the enclave's perspective.